src/cascadia/TerminalApp/Pane.cpp

```diff
@@ -0,0 +1,486 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "Pane.h"
+
+using namespace winrt::Windows::UI::Xaml;
+using namespace winrt::Windows::UI::Core;
+using namespace winrt::Microsoft::Terminal::Settings;
+using namespace winrt::Microsoft::Terminal::TerminalControl;
+
+static const int SEPERATOR_SIZE = 4;
+
+Pane::Pane(GUID profile, TermControl control, const bool lastFocused) :
+    _control{ control },
+    _lastFocused{ lastFocused },
+    _profile{ profile },
+    _splitState{ SplitState::None },
+    _firstChild{ nullptr },
+    _secondChild{ nullptr },
+    _connectionClosedToken{},
+    _firstClosedToken{},
+    _secondClosedToken{}
+{
+    _root = Controls::Grid{};
+    _AddControlToRoot(_control);
+
+    // Set the background of the pane to match that of the theme's default grid
+    // background. This way, we'll match the small underline under the tabs, and
+    // the UI will be consistent on bot light and dark modes.
+    auto res = Application::Current().Resources();
+    winrt::Windows::Foundation::IInspectable key = winrt::box_value(L"BackgroundGridThemeStyle");
+    if (res.HasKey(key))
+    {
+        winrt::Windows::Foundation::IInspectable g = res.Lookup(key);
+        winrt::Windows::UI::Xaml::Style style = g.try_as<winrt::Windows::UI::Xaml::Style>();
+        _root.Style(style);
+    }
+}
+
+Pane::~Pane()
+{
+}
+
+// Method Description:
+// - Adds a given Terminal Control to our root Grid. Also registers an event
+//   handler to know when that control closed.
+// Arguments:
+// - control: The new TermControl to use as the content of our root Grid.
+void Pane::_AddControlToRoot(TermControl control)
+{
+    _root.Children().Append(control.GetControl());
+
+    _connectionClosedToken = control.ConnectionClosed([=]() {
+        if (control.CloseOnExit())

```

Comment:
I'm not 100% on how i feel about asking the control to tell us if it's close on exit because _we knew when we made it_

Reply:
We knew when we made it, but it might have changed since then. I didn't want to propagate another bool around the tree of panes, and update that when the settings reload, if I could instead just ask the control.